### DISCUSSION TITLE
Setting to prevent conversion between TIMESTAMP and TIMESTAMPTZ based on system context

### DISCUSSION THREAD

**Discussion by soerenwolfers** [duckdb/duckdb#11570] [Ideas]
The fact that TIMESTAMP to TIMESTAMPTZ casts use the system time zone can cause ugly problems. I think it's best practice to use `AT TIMEZONE` casts and would be happy if I could force myself to via a configuration option.

(I know that I can already set the timezone used by duckdb for these casts, but that's not what I am looking for.)

**Comment by Tishj** [comment]
Why is that not what you're looking for?
What is the difference in behavior of using `set timezone='UTC'` and your proposed solution?

**Reply by soerenwolfers** [reply]
(1) Because I'm fine with using system time to display TIMESTAMPTZ (since the timezone is included  in these displays). Then again, I wouldn't want TIMESTAMPTZ::VARCHAR casts to assume timezones... 

(2) Because ultimately I want to give people in my team including myself after not touching duckdb for a few weeks) a "can't shoot yourself in the foot" setup for duckdb. In my experience, it's just best to be explicit about things and even though `timezone='UTC' is sometimes what I want, I sometimes really want to be presented from thoughtlessly making a conversion where u should really stop and be explicit about what timezone is meant. 

(3) The fact that PostgreSQL has different behavior on string-with-timezone-component to TIMESTAMP conversion, even with `timezone='UTC'` makes this area dangerous too and I'd like for as little implicitness to happen as possible:
```sql
'2020-01-01 00:00:00+13:00'::TIMESTAMP --- returns 2019-12-31 11:00:00 in duckdb but 2020-01-01T00:00:00.000Z in PostgreSQL
```

(To avoid this and other PostgreSQL discrepancy I think I'd even want more things changed but I have to admit I'm not sure yet what I want there.) 

---

Just to be clear, duckdb's timestamp handling is already a ton better than everything else I used before where I'm now using duckdb, and I mostly run into these issues and questions when I work with datasets that have been created with other tools and have crazy or plain wrong timestamps in sometimes exotic or plain wrong storage representation.

**Reply by soerenwolfers** [reply]
(See other tickets/discussions in the no-footguns-setup series:
https://github.com/duckdb/duckdb/issues/11520
https://github.com/duckdb/duckdb/issues/11521
https://github.com/duckdb/duckdb/discussions/11539
https://github.com/duckdb/duckdb/discussions/11545 
https://github.com/duckdb/duckdb/discussions/10956

Sorry if I'm being ignorant and calling things footguns just because I haven't grokked them
)

**Comment by szarnyasg** [comment]
cc @hawkfish

**Reply by hawkfish** [reply]
Not sure the current casting infrastructure can support disabling specific casts in a context. We would have to be able to stash them somewhere and then stick them back in. And what if an extension gets loaded that adds new ones? This is the old tail patching nightmare...

**Comment by soerenwolfers** [comment]
fixed by https://github.com/duckdb/duckdb/commit/88c4366deacd3c98b6f3c54ebe3306552a374877
